onnxruntime/core/providers/cpu/tensor/quantize_linear.cc

```diff
@@ -47,17 +63,9 @@ Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
     ORT_ENFORCE(x_scale.Shape().NumDimensions() == 1 && x_scale.Shape().Size() == broadcast_dim,
                 "x_scale must be 1D tensor with size ",
                 broadcast_dim);
-    ORT_ENFORCE(x_zero_point != nullptr && x_zero_point->Shape().NumDimensions() == 1 && x_zero_point->Shape().Size() == broadcast_dim,
-                "x_zero_point must be 1D tensor with size ",
+    ORT_ENFORCE(x_zero_point == nullptr || (x_zero_point->Shape().NumDimensions() == 1 && x_zero_point->Shape().Size() == broadcast_dim),
```

Comment:
could this just be x_zero_point->Shape()[0] == broadcast_dim?

Reply:
This would apply to the other cases where Shape().Size() is calculated too. Mostly caring about code size here.

Reply:
x_zero_point->Shape()[0] may not exists even if x_zero_point is not null